app/src/main/java/com/example/lhi06/residue/FindAlbumActivity.java

```diff
@@ -1,36 +1,66 @@
 package com.example.lhi06.residue;
 
+import android.os.Bundle;
 import android.support.annotation.NonNull;
 import android.support.v7.app.AppCompatActivity;
-import android.os.Bundle;
 import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.widget.EditText;
 
 import com.squareup.picasso.Picasso;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.io.IOException;
 
 import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class FindAlbumActivity extends AppCompatActivity {
 
     AlbumAdapter adapter;
+    ItunesService service;
 
-    private List<Album> listAlbum = new ArrayList<>();
 
     @BindView(R.id.recycleView) RecyclerView recyclerView;
+    @BindView(R.id.request) EditText editText;
 
     @Override
     protected void onCreate(@NonNull final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_find_album);
+        ButterKnife.bind(this);
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
         adapter = new AlbumAdapter(Picasso.with(this));
-        // populateRecyclerViewValues():
+        recyclerView.setAdapter(adapter);
+        service = new ItunesService();
+
     }
 
-    private void populateRecyclerViewValues() {
-        // TODO: API buidler and Album list Populater
+    @OnClick(R.id.search_button)
+    public void onSearchClicked() {
+        try {
+            service.getArtistsAlbums(editText.getText().toString()).enqueue(new Callback<AlbumSearchResponse>() {
+                @Override
+                public void onResponse(final Call<AlbumSearchResponse> call, final Response<AlbumSearchResponse> response) {
+                    // on background thread
+                    adapter.setListContent(response.body().getResults());
+                }
+
+                @Override
+                public void onFailure(final Call<AlbumSearchResponse> call, final Throwable t) {
+                    Log.i("LEON", "onFailure:  FAILED" + t);
+                }
+            });
+
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Something like:

            Toast.makeText(this, R.string.search_error_message, Toast.LENGTH_SHORT).show();

It will pop up a message to the user which will disappear after a few seconds